src/codegen/table.cpp

```diff
@@ -91,11 +91,11 @@ void Table::DoGenerateScan(CodeGen &codegen, llvm::Value *table_ptr,
 
     // Generate the scan cover over the given tile group
     if (vector_size > 1) {
-      tile_group_.GenerateVectorizedTidScan(
-          codegen, tile_group_ptr, column_layouts, vector_size, consumer);
+      tile_group_.GenerateVectorizedTidScan(codegen, tile_group_idx,
```

Comment:
Can we propagate the ID through the ScanConsumer callback hooks [here](https://github.com/cmu-db/peloton/blob/master/src/include/codegen/scan_consumer.h#L36)?